compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2522,7 +2522,9 @@ impl<'tcx> TyCtxt<'tcx> {
             && if self.features().collapse_debuginfo {
                 span.in_macro_expansion_with_collapse_debuginfo()
```

Comment:
Could you check that the behaviour is correct with `#![feature(collapse_debuginfo)]` enabled?

Reply:
Yeah, this already works correctly because `in_macro_expansion_with_collapse_debuginfo` already checks the `kind` field for `ExpnKind::Macro` while MIR inlining generates `ExpnKind::Inlined`